File: Rust/reqwest.rs
use reqwest::blocking::Client;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
    let client = Client::new();

    let response = client
        .get("https://api.apitube.io/v1/news/everything")
        .query(&[("per_page", "50"), ("api_key", "YOUR_API_KEY")])
        .send()?;

    if response.status().is_success() {
        let body = response.text()?;
        println!("{}", body);
    } else {
        eprintln!("HTTP Request Error: {}", response.status());
    }

    Ok(())
}

File: Ruby/Reqwest.rb
use reqwest;

#[tokio::main]
async fn main() -> Result<(), reqwest::Error> {
    make_request().await?;
    Ok(())
}

async fn make_request() -> Result<(), reqwest::Error> {
    let url = "https://api.apitube.io/v1/news/everything?per_page=50&api_key=YOUR_API_KEY";

    let response = reqwest::get(url).await?;

    match response.status().as_u16() {
        200..=299 => {
            let body = response.text().await?;
            println!("Response: {}", body);
        }
        400..=499 => {
            println!("Client Error: {}", response.status());
        }
        500..=599 => {
            println!("Server Error: {}", response.status());
        }
        _ => {
            println!("HTTP Request Error: Unexpected status code: {}", response.status());
        }
    }

    Ok(())
}
